resolver/GlobalPass.cc

```diff
@@ -82,7 +82,12 @@ bool resolveTypeMember(core::GlobalState &gs, core::ClassOrModuleRef parent, cor
     if (!sym.data(gs)->derivesFrom(gs, core::Symbols::Class()) && myVariance != parentVariance &&
         myVariance != core::Variance::Invariant) {
         if (auto e = gs.beginError(myTypeMember.data(gs)->loc(), core::errors::Resolver::ParentVarianceMismatch)) {
-            e.setHeader("Type variance mismatch with parent `{}`", parent.show(gs));
+            e.setHeader(
+                "Type variance mismatch with parent `{}`; child `{}` should be `{}` or invariant, but it is `{}`",
+                parent.show(gs), sym.show(gs), core::Polarities::showVariance(parentVariance),
+                core::Polarities::showVariance(myVariance));
+            e.addErrorLine(parentTypeMember.data(gs)->loc(), "Variance for parent `{}` declared here", parent.show(gs));
+            e.addErrorNote("Sorbet requires compatible variance between parent and child in order to type check.");
```

Comment:
I don't think this error note is adding much value (it doesn't say anything new that isn't said earlier). For example, we could have a note like this on almost every error in Sorbet ("Sorbet requires compatible argument types in order to type check" / "Sorbet requires compatible return types in order to type check" / etc.)

My suggestion might be to instead direct people to read the error documentation for this error code (5015), as there is an extended example outlining why this "compatible variance" check is important.